openscenario/openscenario_utility/openscenario_utility/validation.py

```diff
@@ -18,10 +18,54 @@
 from pathlib import Path
 from pkg_resources import resource_string
 from sys import exit
+from openscenario_utility.conversion import from_yaml, load_yaml
 
 import argparse
 import xmlschema
 
+def find_in_dict(key, dictionary):
+    values = list()
+    if key in dictionary:
+        values.append(dictionary[key])
+    for k, v in dictionary.items():
+        if k == key:
+            values.append(v)
+        if isinstance(v, list):
+            for i in v:
+                for j in find_in_dict(key, i):
+                    values.append(j)
+        elif isinstance(v,dict):
+            for result in find_in_dict(key, v):
+                values.append(result)
+    return values
+
+class ScenarioValidator:
+    def __init__(self):
+        self.warning_message = ''
+
+    def __call__(self, path: Path):
+        return None
+
+    def get_warning_message(self):
+        return self.warning_message
+
+
+class ReachPositionConditionValidator(ScenarioValidator):
+    def __init__(self):
+        self.warning_message = "ReachPositionCondition position does not match the last position of the path"
+        self.schema = xmlschema.XMLSchema(
+            resource_string(__name__, "resources/OpenSCENARIO.xsd").decode("utf-8")
+        )
+
+    def __call__(self, path: Path):
+        scenario = self.schema.to_dict(str(path))
+        #Search for the last Routing Action destionation:
+        last_destination = find_in_dict('Position',find_in_dict('RoutingAction', scenario)[-1])
+
+        #Search for ReachPositionCondition:
+        reach_position_conditon = find_in_dict('Position',find_in_dict('ReachPositionCondition', scenario)[-1])
+
+        return reach_position_conditon == last_destination
```

Comment:
Isn't it too strict? Shouldn't we assure that the distance between `reach_position_conditon` and `last_destination` is lower than some kind of threshold (maybe `tolerance` from scenario)?